src/simulator.zig

```diff
@@ -196,6 +203,14 @@ fn chance(random: *std.rand.Random, p: u8) bool {
     return random.uintAtMost(u8, 100) <= p;
 }
 
+/// Returns the next argument for the simulator or null (if none available)
+fn args_next(args: *std.process.ArgIterator, allocator: *std.mem.Allocator) ?[:0]const u8 {
+    return if (args.next(allocator)) |err_or_bytes|
+        err_or_bytes catch @panic("Unable to extract next value from args")
+    else
+        null;
```

Comment:
I think the version using `orelse` is cleaner here as well. What do you think?
```suggestion
    const err_or_bytes = args.next(allocator) orelse return null;
    return err_or_bytes catch @panic("Unable to extract next value from args");
```